### PR TITLE
feat: data subscriber, evaluation round

### PR DESCRIPTION
**Pull request by elijahmg**

### CODE REVIEW

File: src/local-services/use-local-websocket-server.ts [hunk 1]
```diff
@@ -3,7 +3,12 @@ import { useEffect } from 'react';
 const WEBSOCKET_SERVER_URL = 'ws://localhost:5001';
 
 // @TODO fix payload type
-export function useLocalWebsocketServer(callback: (payload: any) => void) {
+export function useLocalWebsocketServer(
+  callback: (payload: any) => void,
+  deps?: any[],
```
Comment (martinmachma) [inline src/local-services/use-local-websocket-server.ts:8]: Maybe change any to unknown?

File: src/pages/[quizId]/overview/index.tsx [hunk 2]
```diff
@@ -46,6 +47,7 @@ const Overview = () => {
 };
 
 Overview.getLayout = function getLayout(pageContent: React.ReactElement) {
+  // InQuizWrapper has prop to pass custom quiz status enum value
```
Comment (martinmachma) [inline src/pages/[quizId]/overview/index.tsx:50]: Why is this comment here?

File: src/pages/[quizId]/play.tsx [hunk 3]
```diff
@@ -23,6 +26,15 @@ const Play = () => {
     teamData: state.teamData,
   }));
 
+  // for update of the quiz status through websocket
+  useSubscribeToQuizTeamUpdateStore();
```
Comment (martinmachma) [inline src/pages/[quizId]/play.tsx:30]: Wouldnt it be nicer to have that subscription directly in the `useTeamQuizDataStore` hook?

File: src/components/wrappers/in-quiz-wrapper.tsx [hunk 4]
```diff
@@ -1,13 +1,48 @@
...
+  useEffect(() => {
+    if (!quizData.pin) {
+      router.push('/pin');
+    }
+  }, []);
+
+  trpc.team.getQuizByPin.useQuery(
+    {
+      pin: quizData.pin!,
+      quizStatus: quizStatus || QuizStatusEnum.PLAYING,
```
Comment (martinmachma) [inline src/components/wrappers/in-quiz-wrapper.tsx:36]: Why is it not enough to just query by pin?

File: src/pages/admin/[quizId]/quiz-control.tsx [hunk 5]
```diff
@@ -38,13 +48,37 @@ const QuizControl = () => {
...
+  const handleEndRound = async () => {
+    if (!quizData.id) return;
+
+    setIsRoundEnded(true);
+
+    await updateQuizStatus({
+      id: quizData.id,
+      quizStatus: QuizStatusEnum.END_ROUND,
+    });
+  };
```
Comment (martinmachma) [inline src/pages/admin/[quizId]/quiz-control.tsx:60]: I think async/await is not needed

**Review by martinmachma — APPROVED** [review]

File: src/local-services/use-local-websocket-server.ts (hunk 1 above)
Comment (elijahmg) [reply src/local-services/use-local-websocket-server.ts:8]: I'm fan of any 😄

File: src/pages/[quizId]/overview/index.tsx (hunk 2 above)
Comment (elijahmg) [reply src/pages/[quizId]/overview/index.tsx:50]: Just as a reminder

File: src/components/wrappers/in-quiz-wrapper.tsx (hunk 4 above)
Comment (elijahmg) [reply src/components/wrappers/in-quiz-wrapper.tsx:36]: BE called is reused in couple of places, Initially I wanted to have it only for pin screen, where you get quiz which is in state of Joining, but during this PR decided to make it a bit more general, to have status more flexible.
And to answer your question, you can imagine that there will be more quizzes with the same pin but different statuses. I know It does not solve uniqueness problem completely, but for now it's good enough

File: src/pages/admin/[quizId]/quiz-control.tsx (hunk 5 above)
Comment (elijahmg) [reply src/pages/admin/[quizId]/quiz-control.tsx:60]: You're right

File: src/pages/[quizId]/play.tsx (hunk 3 above)
Comment (elijahmg) [reply src/pages/[quizId]/play.tsx:30]: I thought about it. I'll need to check usage again

**Comment by elijahmg** [comment]
@martinmachma 
I'll merge for now, but will keep in mind your comments about `useSubscribeToQuizTeamUpdateStore `

**Pull request merged**